tests/counters.rs

```diff
@@ -0,0 +1,148 @@
+// These tests cannot be run in parallel. We use the "serial_test" crate to run
+// them one by one.
+
+mod helpers;
+
+#[cfg(test)]
+mod counters {
+    use crate::helpers::*;
+    use http::StatusCode;
+    use infinispan::request::counters;
+    use reqwest::Response;
+    use serial_test::serial;
+    use std::collections::HashSet;
+    use std::iter::FromIterator;
+
+    #[tokio::test]
+    #[serial]
+    async fn create_weak_counter() {
+        cleanup().await;
+
+        let counter_name = "test_counter";
+
+        run(counters::create_weak(counter_name)).await;
+        let resp = run(counters::get(counter_name)).await;
```

Comment:
Did you forget about handling the response status of the first `run()` call? In this case where a function is called more than once with values ignored and values not ignored, I'd use:
```rust
let _ = run(counters::create_weak(counter_name)).await;
let resp = run(counters::get(counter_name)).await;
```
Note that Rust won't yell at you for directly ignoring the value in the first call because while `Result` has a `#[must_use]` attribute, the `Response` type does not have it, and you unwrap the result unconditionally in the `run` helper.

My suggestion is using `let _ = run(...).await?;`,have these test fns return `Result<(), reqwest::Error>` or some other error type usable with `?` with a conversion for `reqwest::Error` (in order to avoid the boilerplate of manually converting it in each call), while removing the `unwrap()` from `run()` and adjusting the return type to `Result<Response, reqwest::Error>` which will enforce you make use of the result value or explicitly ignore it.

As an additional benefit to being explicit about what you do with the returned values, I *think* failures in `run().await?;` will generate a nicer error report for you to debug.

Reply:
I did not forget about handling the response in the first call. If the `create_weak()` call fails it's because it couldn't reach Infinispan. In that case, it's fine to just abort the test there with the `unwrap` in `run()`.

Also, I don't see the benefit of using `let _` here :thinking: . I think the test states its intention: "call create_weak() and continue unless it cannot contact Infinispan". We don't care what it returns, because we are going to check that the counter exists with the `get` call.

Another option is to check the response of `create_weak` and don't call `get`. We could check the status code returned for example. But in my opinion, that's not as good.

Reply:
> I did not forget about handling the response in the first call. If the `create_weak()` call fails it's because it couldn't reach Infinispan. In that case, it's fine to just abort the test there with the `unwrap` in `run()`.

The (minor) problem is _I_ don't know and need to check the logic to make sure, and using `let _` is free and signals that you didn't forget.

If the `run()` call fails, it is very likely it is because it couldn't reach infinispan, and it is ok to abort there. Unless it isn't or you don't know why it panicked there, and then need to go over a backtrace several levels deep to find out. I know this is again minor, but tests and tested code change, and different conditions could trigger an error, especially with a network, where it could be the service is not up and easy to find out in a CI system, but it could also be "file descriptors exhausted" or some other condition. If you could get more complete information from these helpers it could well pay off in the long run.

This is all minor, so once you have it figured out, :+1: from me!